lib/job-iteration/iteration.rb

```diff
@@ -176,7 +176,13 @@ def assert_implements_methods!
         )
       end
 
-      unless respond_to?(:build_enumerator, true)
+      if respond_to?(:build_enumerator, true)
+        parameters = method(:build_enumerator).parameters
+        unless valid_cursor_parameter?(parameters)
+          raise ArgumentError, "Iteration job (#{self.class}) #build_enumerator " \
+          "expects the keyword argument `cursor` please make sure that this condition is met"
```

Comment:
Nitpick: `please make sure that this condition is met` may not the language that's typically used in exception messages. "expects the keyword argument `cursor`" is probably enough for developers to understand that's a condition that they have to satisfy :)